mungegithub/mungers/approvers/approvers_test.go

```diff
@@ -0,0 +1,386 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package approvers
+
+import (
+	"testing"
+
+	"fmt"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"reflect"
+)
+
+func TestUnapprovedFiles(t *testing.T) {
+	rootApprovers := sets.NewString("Alice", "Bob")
+	aApprovers := sets.NewString("Art", "Anne")
+	bApprovers := sets.NewString("Bill", "Ben", "Barbara")
+	cApprovers := sets.NewString("Chris", "Carol")
+	dApprovers := sets.NewString("David", "Dan", "Debbie")
+	eApprovers := sets.NewString("Eve", "Erin")
+	edcApprovers := eApprovers.Union(dApprovers).Union(cApprovers)
+	FakeRepoMap := map[string]sets.String{
+		"":        rootApprovers,
+		"a":       aApprovers,
+		"b":       bApprovers,
+		"c":       cApprovers,
+		"a/d":     dApprovers,
+		"a/combo": edcApprovers,
+	}
+	tests := []struct {
+		testName           string
+		filenames          []string
+		currentlyApproved  sets.String
+		expectedUnapproved sets.String
+	}{
+		{
+			testName:           "Empty PR",
+			filenames:          []string{},
+			currentlyApproved:  sets.NewString(),
+			expectedUnapproved: sets.NewString(),
+		},
+		{
+			testName:           "Single Root File PR Approved",
+			filenames:          []string{"kubernetes.go"},
+			currentlyApproved:  sets.NewString(rootApprovers.List()[0]),
+			expectedUnapproved: sets.NewString(),
+		},
+		{
+			testName:           "Single Root File PR No One Approved",
+			filenames:          []string{"kubernetes.go"},
+			currentlyApproved:  sets.NewString(),
+			expectedUnapproved: sets.NewString(""),
```

Comment:
yeah, but that's what github sends us back.  E.g the makefile in the root dir is listed as `Makefile` not `/Makefile`